SIL.Archiving/ArchivingDlgViewModel.cs

```diff
@@ -268,41 +323,87 @@ await Task.Run(() =>
 		{
 			Guard.AgainstNull(groupId, nameof(groupId));
 			if (FileLists.ContainsKey(groupId))
 				throw new ArgumentException("Duplicate file group ID: " + groupId, nameof(groupId));
-			FileLists[groupId] = Tuple.Create(files, progressMessage);
+			FileLists[groupId] = Tuple.Create(files, addingToArchiveProgressMessage);
 		}
 
 		/// ------------------------------------------------------------------------------------
 		private void DisplayInitialSummary(CancellationToken cancellationToken)
 		{
+			Console.WriteLine($"{ArchiveType} Tests TEMP: {nameof(DisplayInitialSummary)} start");
+
 			if (OverrideDisplayInitialSummary != null)
-				OverrideDisplayInitialSummary(FileLists, cancellationToken);
-			else
 			{
-				ReportProgress(Progress.GetMessage(StringId.PreArchivingStatus), MessageType.Normal, cancellationToken);
+				OverrideDisplayInitialSummary(FileLists, cancellationToken);
+				return;
+			}
+
+			Console.WriteLine($"{ArchiveType} Tests TEMP: {nameof(DisplayInitialSummary)} after override");
+
+			foreach (var message in AdditionalMessages)
+				DisplayMessage(message.Key + "\n", message.Value);
 
-				if (OnReportMessage != null)
+			Console.WriteLine($"{ArchiveType} Tests TEMP: {nameof(DisplayInitialSummary)} after additional");
+
+			if (GetOverriddenPreArchivingMessages != null)
+			{
+				bool firstMsg = true;
+				foreach (var msg in GetOverriddenPreArchivingMessages(FileLists))
 				{
-					foreach (var kvp in FileLists)
+					if (firstMsg)
 					{
-						string msg = FileGroupDisplayMessage(kvp.Key);
-						if (msg != Empty)
-							OnReportMessage(msg, MessageType.Indented);
-
-						foreach (var file in kvp.Value.Item1)
-							OnReportMessage(Path.GetFileName(file), MessageType.Bullet);
+						ReportProgress(msg.Item1, msg.Item2, cancellationToken);
+						firstMsg = false;
 					}
+					else
+						DisplayMessage(msg.Item1, msg.Item2);
 				}
+
+				Console.WriteLine($"{ArchiveType} Tests TEMP: {nameof(DisplayInitialSummary)} after overridden Pre-Archiving messages");
+			}
+			else
+			{
+				ReportProgress(Progress.GetMessage(StringId.PreArchivingStatus),
+					MessageType.Normal, cancellationToken);
+
+				Console.WriteLine($"{ArchiveType} Tests TEMP: {nameof(DisplayInitialSummary)} after normal Pre-Archiving message");
 			}
+
+			foreach (var kvp in FileLists)
+			{
+				if (cancellationToken.IsCancellationRequested)
+					throw new OperationCanceledException();
+
+				string msg = FileGroupDisplayMessage(kvp.Key);
+				if (msg != Empty)
+					DisplayMessage(msg, InitialFileGroupDisplayMessageType);
+
+				foreach (var file in kvp.Value.Item1)
+					DisplayMessage(Path.GetFileName(file), MessageType.Bullet);
+			}
+
+			Console.WriteLine($"{ArchiveType} Tests TEMP: {nameof(DisplayInitialSummary)} end");
 		}
 
 		protected virtual string FileGroupDisplayMessage(string groupKey)
 		{
-			return groupKey;
+			return OverrideGetFileGroupDisplayMessage == null ? groupKey : OverrideGetFileGroupDisplayMessage(groupKey);
```

Comment:
since `OverrideGetFileGroupDisplayMessage` could return null we should do `OverrideGetFileGroupDisplayMessage?.Invoke(groupKey) ?? groupKey` instead, then we will not return null unless groupKey was null, which would have been the case prior to this change.

Reply:
Done